crates/uv/src/settings.rs

```diff
@@ -2452,30 +2457,33 @@ impl ResolverSettings {
 
 impl From<ResolverOptions> for ResolverSettings {
     fn from(value: ResolverOptions) -> Self {
+        let index_locations = IndexLocations::new(
+            value
+                .index
+                .into_iter()
+                .flatten()
+                .chain(value.extra_index_url.into_iter().flatten().map(Index::from))
+                .chain(value.index_url.into_iter().map(Index::from))
+                .collect(),
+            value
+                .find_links
+                .into_iter()
+                .flatten()
+                .map(Index::from)
+                .collect(),
+            value.no_index.unwrap_or_default(),
+        );
+        let url_auth_policies = (&index_locations).into();
```

Comment:
In general (not a hard rule) I have a mild preference for `From` because it makes the types more obvious. For example, as a reader here, I actually don't know what type this is being converted into? Whereas `UrlAuthPolicies::from` tells you that immediately.